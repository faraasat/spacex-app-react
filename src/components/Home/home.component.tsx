import { Link } from "react-router-dom";
import "./home.styles.css";
import Galaxy1 from "../../assets/galaxy1.png";

const HomeComponent = () => {
  return (
    <div>
      <div className="background-container">
        <img
          src="https://s3-us-west-2.amazonaws.com/s.cdpn.io/1231630/moon2.png"
          alt=""
        />
        <div className="stars"></div>
        <div className="twinkling"></div>
        <div className="clouds"></div>
      </div>
      <div className="home-typo">
        <h1>
          SpaceX<span> App</span>
        </h1>
        <p>
          Lorem ipsum dolor sit amet consectetur adipisicing elit. Minus sed
          deleniti porro veniam at dolores qui soluta nulla perferendis voluptas
          iure ipsa doloremque aspernatur, tempora harum quam voluptatum
          explicabo et?
        </p>
        <button>
          <Link to="/explore">Get Started!</Link>
        </button>
      </div>
      <div className="objects">
        <img
          className="object_rocket"
          src="http://salehriaz.com/404Page/img/rocket.svg"
          width="40px"
          alt="rocket"
        />
        <img
          className="object_moon"
          src="http://salehriaz.com/404Page/img/moon.svg"
          width="80px"
          alt="small moon"
        />
        <img className="object-galaxy__1" src={Galaxy1} alt="galaxy 1" />
        <div className="box_astronaut">
          <img
            className="object_astronaut"
            src="http://salehriaz.com/404Page/img/astronaut.svg"
            width="140px"
            alt="astronaut"
          />
        </div>
      </div>
    </div>
  );
};

export default HomeComponent;
