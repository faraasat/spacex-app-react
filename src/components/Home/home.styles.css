@import url("https://fonts.googleapis.com/css?family=Dosis:300,400,500");

@keyframes move-background {
  from {
    -webkit-transform: translate3d(0px, 0px, 0px);
  }
  to {
    -webkit-transform: translate3d(1000px, 0px, 0px);
  }
}
@-webkit-keyframes move-background {
  from {
    -webkit-transform: translate3d(0px, 0px, 0px);
  }
  to {
    -webkit-transform: translate3d(1000px, 0px, 0px);
  }
}

@-moz-keyframes move-background {
  from {
    -webkit-transform: translate3d(0px, 0px, 0px);
  }
  to {
    -webkit-transform: translate3d(1000px, 0px, 0px);
  }
}

@-webkit-keyframes move-background {
  from {
    -webkit-transform: translate3d(0px, 0px, 0px);
  }
  to {
    -webkit-transform: translate3d(1000px, 0px, 0px);
  }
}

.background-container {
  position: fixed;
  top: 0;
  left: 0;
  bottom: 0;
  right: 0;
  z-index: -5;
}

.stars {
  background: black
    url(https://s3-us-west-2.amazonaws.com/s.cdpn.io/1231630/stars.png) repeat;
  position: absolute;
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
  display: block;
  z-index: 0;
}

.twinkling {
  width: 10000px;
  height: 100%;
  background: transparent
    url("https://s3-us-west-2.amazonaws.com/s.cdpn.io/1231630/twinkling.png")
    repeat;
  background-size: 1000px 1000px;
  position: absolute;
  right: 0;
  top: 0;
  bottom: 0;
  z-index: 2;

  -moz-animation: move-background 70s linear infinite;
  -ms-animation: move-background 70s linear infinite;
  -o-animation: move-background 70s linear infinite;
  -webkit-animation: move-background 70s linear infinite;
  animation: move-background 70s linear infinite;
}

.clouds {
  width: 10000px;
  height: 100%;
  background: transparent
    url("https://s3-us-west-2.amazonaws.com/s.cdpn.io/1231630/clouds_repeat.png")
    repeat;
  background-size: 1000px 1000px;
  position: absolute;
  right: 0;
  top: 0;
  bottom: 0;
  z-index: 3;

  -moz-animation: move-background 150s linear infinite;
  -ms-animation: move-background 150s linear infinite;
  -o-animation: move-background 150s linear infinite;
  -webkit-animation: move-background 150s linear infinite;
  animation: move-background 150s linear infinite;
}

.background-container img {
  height: 70vh;
  width: 70vh;
  position: absolute;
  z-index: 3;
  right: 20px;
  top: 70px;
  animation: spin 600s infinite linear;
}

@keyframes spin {
  0% {
    -webkit-transform: rotate(0deg);
  }
  100% {
    -webkit-transform: rotate(360deg);
  }
}

.home-typo {
  position: absolute;
  z-index: 500;
  display: flex;
  align-items: flex-start;
  justify-content: center;
  flex-direction: column;
  width: 100%;
  height: calc(100vh - 64px);
}

.home-typo h1 {
  width: 100%;
  color: rgb(29, 142, 248);
  margin-left: 10%;
  font-size: 85px;
  margin-block: 0em;
  z-index: 500;
}

.home-typo h1 span {
  color: rgb(231, 231, 231);
  font-size: 75px;
  margin-block: 0em;
  z-index: 500;
}

.home-typo p {
  color: rgb(190, 190, 190);
  width: 40%;
  margin-left: 10%;
  font-size: 20px;
  z-index: 500;
}

.home-typo button {
  margin-left: 10%;
  padding: 14px 49px;
  background-color: orange;
  border: none;
  outline: none;
  cursor: pointer;
  border-radius: 5px;
  transition: all 0.3s ease-in-out;
}

.home-typo button a {
  text-decoration: none;
  font-size: 20px;
  font-weight: bold;
  color: aliceblue;
}

.home-typo button:hover {
  transform: scale(1.05);
  box-shadow: 0px 0px 5px 1px rgba(0, 140, 255, 0.5);
}

.home-typo button:active {
  transform: scale(0.99);
  box-shadow: 0px 0px 5px 1px rgba(0, 140, 255, 0.65);
}

@-moz-keyframes rocket-movement {
  100% {
    -moz-transform: translate(1200px, -600px);
  }
}
@-webkit-keyframes rocket-movement {
  100% {
    -webkit-transform: translate(1200px, -600px);
  }
}
@keyframes rocket-movement {
  100% {
    transform: translate(1200px, -600px);
  }
}
@-moz-keyframes move-astronaut {
  100% {
    -moz-transform: translate(-160px, -160px);
  }
}
@-webkit-keyframes move-astronaut {
  100% {
    -webkit-transform: translate(-160px, -160px);
  }
}
@keyframes move-astronaut {
  100% {
    -webkit-transform: translate(-160px, -160px);
    transform: translate(-160px, -160px);
  }
}
@-moz-keyframes rotate-astronaut {
  100% {
    -moz-transform: rotate(-720deg);
  }
}
@-webkit-keyframes rotate-astronaut {
  100% {
    -webkit-transform: rotate(-720deg);
  }
}
@keyframes rotate-astronaut {
  100% {
    -webkit-transform: rotate(-720deg);
    transform: rotate(-720deg);
  }
}

.objects img {
  /* z-index: 90; */
  pointer-events: none;
}

.object_rocket {
  z-index: 1;
  position: absolute;
  transform: translateX(-50px);
  top: 75%;
  pointer-events: none;
  animation: rocket-movement 200s linear infinite both running;
}

.object_moon {
  position: absolute;
  top: 10%;
  left: 10%;
  transform: rotate(0deg);
  transition: transform ease-in 99999999999s;
}

.object_astronaut {
  animation: rotate-astronaut 200s infinite linear both alternate;
}

.box_astronaut {
  z-index: 110 !important;
  position: absolute;
  top: 60%;
  right: 20%;
  will-change: transform;
  animation: move-astronaut 50s infinite linear both alternate;
}

@media only screen and (max-width: 600px) {
  .box_astronaut {
    top: 70%;
  }
  .central-body {
    padding-top: 25%;
  }
}