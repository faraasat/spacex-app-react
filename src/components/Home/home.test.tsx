import { shallow } from "enzyme";
import Home from "./home.component";

describe("Home Component", () => {
  const shallowRender = shallow(<Home />);

  it("should match snapshot", () => {
    expect(shallowRender.debug()).toMatchSnapshot();
  });

  it("should have at least on div", () => {
    expect(shallowRender.find("div").length).toBeGreaterThanOrEqual(1);
  });

  it("should have a className", () => {
    expect(
      shallowRender.find(".background-container").length
    ).toBeGreaterThanOrEqual(1);
  });
});
