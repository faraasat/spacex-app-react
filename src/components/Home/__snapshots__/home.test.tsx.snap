// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Component should match snapshot 1`] = `
"<div>
  <div className=\\"background-container\\">
    <img src=\\"https://s3-us-west-2.amazonaws.com/s.cdpn.io/1231630/moon2.png\\" alt=\\"\\" />
    <div className=\\"stars\\" />
    <div className=\\"twinkling\\" />
    <div className=\\"clouds\\" />
  </div>
  <div className=\\"home-typo\\">
    <h1>
      SpaceX
      <span>
         App
      </span>
    </h1>
    <p>
      Lorem ipsum dolor sit amet consectetur adipisicing elit. Minus sed deleniti porro veniam at dolores qui soluta nulla perferendis voluptas iure ipsa doloremque aspernatur, tempora harum quam voluptatum explicabo et?
    </p>
    <button>
      <Link to=\\"/explore\\">
        Get Started!
      </Link>
    </button>
  </div>
  <div className=\\"objects\\">
    <img className=\\"object_rocket\\" src=\\"http://salehriaz.com/404Page/img/rocket.svg\\" width=\\"40px\\" alt=\\"rocket\\" />
    <img className=\\"object_moon\\" src=\\"http://salehriaz.com/404Page/img/moon.svg\\" width=\\"80px\\" alt=\\"small moon\\" />
    <img className=\\"object-galaxy__1\\" src=\\"galaxy1.png\\" alt=\\"galaxy 1\\" />
    <div className=\\"box_astronaut\\">
      <img className=\\"object_astronaut\\" src=\\"http://salehriaz.com/404Page/img/astronaut.svg\\" width=\\"140px\\" alt=\\"astronaut\\" />
    </div>
  </div>
</div>"
`;
